Code/GraphMol/ForceFieldHelpers/Wrap/testHelpers.py

```diff
@@ -135,249 +146,324 @@ def test7(self):
+        self.dirName = os.path.join(
+            RDConfig.RDBaseDir,
+            "Code",
+            "GraphMol",
+            "ForceFieldHelpers",
+            "UFF",
+            "test_data",
+        )
+        fName = os.path.join(self.dirName, "Issue239.mol")
+        m = Chem.MolFromMolFile(fName)
+        self.assertIsNotNone(m)
+        mp = ChemicalForceFields.MMFFGetMoleculeProperties(m)
+        ff = ChemicalForceFields.MMFFGetMoleculeForceField(m, mp)
+        ff.Initialize()
+        positions = ff.Positions()
+        savedPos = list(positions)
+        grad1 = ff.CalcGrad()
+        for v in grad1:
+            self.assertNotAlmostEqual(v, 0.0, 3)
+        ff.Minimize(10000, 1.0e-6, 1.0e-3)
+        grad2 = ff.CalcGrad()
+        for v in grad2:
+            self.assertAlmostEqual(v, 0.0, 3)
+        ff.Initialize()
+        grad2 = ff.CalcGrad(savedPos)
+        self.assertEqual(len(grad1), len(grad2))
+        for i in range(len(grad1)):
+            self.assertAlmostEqual(grad1[i], grad2[i], 3)
+
+    def testGitHub2820(self):
+        m = Chem.MolFromSmiles("[Na]C")
+        self.assertIsNotNone(m)
+        mp = ChemicalForceFields.MMFFGetMoleculeProperties(m)
+        self.assertIsNone(mp)
+        rdDistGeom.EmbedMultipleConfs(m, 2)
+        res = ChemicalForceFields.MMFFOptimizeMoleculeConfs(m)
+        self.assertEqual(len(res), 2)
+        self.assertEqual(res[0], res[1])
+        self.assertEqual(res[0], (-1, -1.0))
+
+    def testOptimizeMolecule(self):
+        m = Chem.AddHs(Chem.MolFromSmiles("CCCO"))
+        self.assertIsNotNone(m)
+        self.assertEqual(rdDistGeom.EmbedMolecule(m), 0)
+        mp = ChemicalForceFields.MMFFGetMoleculeProperties(m)
+        ff = ChemicalForceFields.MMFFGetMoleculeForceField(m, mp)
+        before = ff.CalcEnergy()
+        self.assertEqual(ChemicalForceFields.OptimizeMolecule(ff, maxIters=200), 0)
+        after = ff.CalcEnergy()
+        self.assertLess(after, before)
+
+    def testOptimizeMoleculeConfs(self):
+        m = Chem.AddHs(Chem.MolFromSmiles("CCCO"))
+        self.assertIsNotNone(m)
+        cids = rdDistGeom.EmbedMultipleConfs(m, numConfs=10)
+        self.assertEqual(len(cids), 10)
+        mp = ChemicalForceFields.MMFFGetMoleculeProperties(m)
+        ff = ChemicalForceFields.MMFFGetMoleculeForceField(m, mp)
+        before = [
+            ChemicalForceFields.MMFFGetMoleculeForceField(
+                m, mp, confId=cid
+            ).CalcEnergy()
+            for cid in cids
+        ]
+        res, after = tuple(
+            zip(*ChemicalForceFields.OptimizeMoleculeConfs(m, ff, maxIters=200))
+        )
+        self.assertEqual(len(res), 10)
+        self.assertEqual(len(before), len(after))
+        self.assertTrue(all(map(lambda i: i == 0, res)))
+        self.assertTrue(all(after[i] < b for i, b in enumerate(before)))
+
+    def testEmptyFF(self) -> None:
+        m = Chem.AddHs(Chem.MolFromSmiles("CCCO"))
+        rdDistGeom.EmbedMolecule(m)
```

Comment:
Instead of embedding the molecule here, please just directly create a molecule with a conformer. You can do this by creating a CXSMILES for a molecule that already has a conformer and constructing amolecule here directly from that.
Something like:
```
m = Chem.MolFromSmiles('CCCO |(-1.28533,-0.0567758,0.434662;-0.175447,0.695786,-0.299881;0.918409,-0.342619,-0.555572;1.30936,-0.801512,0.71705)|')
```
I don't think you need the Hs for this test, so it can be simple.